velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -1111,51 +1149,16 @@ void SubstraitVeloxPlanConverter::separateFilters(
   }
 }
 
-uint32_t SubstraitVeloxPlanConverter::getColumnIndexFromIn(
-    const ::substrait::Expression_ScalarFunction& scalarFunction) {
-  VELOX_CHECK(
-      scalarFunction.arguments().size() == 2,
-      "Two args expected in In expression.");
-  VELOX_CHECK(
-      getExprFromFunctionArgument(scalarFunction.arguments()[0])
-          .has_selection(),
-      "Field expected.");
-
-  // Get the column index.
-  uint32_t colIdx = subParser_->parseReferenceSegment(
-      getExprFromFunctionArgument(scalarFunction.arguments()[0])
-          .selection()
-          .direct_reference());
-  VELOX_CHECK(
-      getExprFromFunctionArgument(scalarFunction.arguments()[1]).has_literal(),
-      "Literal expected.");
-  VELOX_CHECK(
-      getExprFromFunctionArgument(scalarFunction.arguments()[1])
-          .literal()
-          .has_list(),
-      "List expected.");
-  return colIdx;
-}
-
-void SubstraitVeloxPlanConverter::setInValues(
-    const ::substrait::Expression_ScalarFunction& scalarFunction,
-    std::unordered_map<uint32_t, std::shared_ptr<FilterInfo>>& colInfoMap) {
-  // Get the column index.
-  uint32_t colIdx = getColumnIndexFromIn(scalarFunction);
-
-  // Get the value list.
-  std::vector<variant> variants;
-  auto valueList = getExprFromFunctionArgument(scalarFunction.arguments()[1])
-                       .literal()
-                       .list();
-  variants.reserve(valueList.values().size());
-  for (const auto& literal : valueList.values()) {
-    variants.emplace_back(
-        exprConverter_->toTypedVariant(literal)->veloxVariant);
+bool SubstraitVeloxPlanConverter::canPushdownSingularList(
+    const ::substrait::Expression_SingularOrList& singularList) {
+  if (singularList.value().has_selection()) {
+    return true;
+  } else if (singularList.value().has_scalar_function()) {
```

Comment:
When `value` is not field, the IN filter cannot be pushed down. Will fix this in a following PR.